src/logger/index.ts

```diff
@@ -0,0 +1,149 @@
+  'warn': LogLevel.warn
+}
+
+const LOG_LEVEL = config.logger.level && logLevelMapping[config.logger.level] || LogLevel.important;
+const LOG_TO_FILES = config.logger.logToFiles || false;
+let LOG_PATH = config.logger.filePath || path.join(path.resolve(__dirname, '../../'), 'logs/');
+if (!LOG_PATH.endsWith('/')) {
+  LOG_PATH = LOG_PATH + '/';
+}
+
+const ACCESS_FILE_PATH = LOG_PATH + 'access.log';
+const METRICS_FILE_PATH = LOG_PATH + 'metrics.log';
+const EVENTS_FILE_PATH = LOG_PATH + 'events.log';
+const REMOVE_TIMESTAMP = true;
+
+const LOG_FMT = '[:date[web]] ip=:remote-addr ts=:date[iso] method=:method path=:url http_version=:http-version req_header:req[header] status=:status content_length=:res[content-length] content_type=":res[content-type]" ref=:referrer user_agent=:user-agent elapsed_ms=:total-time[3]';
+
+/**
+ * Logs to the console based on log level
+ */
+class ConsoleLogger {
+  public readonly logLevel: LogLevel;
+  private logger: Logger;
+  private includeStackTrace: boolean;
+
+  constructor(logLevel: LogLevel) {
+    this.logger = new Logger(LoggerModes.Console, '', REMOVE_TIMESTAMP);
+    this.logLevel = logLevel;
+    this.includeStackTrace = this.logLevel == LogLevel.debug ? true : false;
+  }
+
+  /**
+   * Calls `this.debug`. Used for stream interfaces.
+   * @param content Content to log
+   */
+  public write(content: string | object): void {
+    this.debug(content);
+  }
+
+  public debug(content: string | object): void {
+    if (this.logLevel > LogLevel.debug) {
+      return;
+    }
+    this.logger.info(content, this.includeStackTrace);
+  }
+
+  public imp(content: string | object): void {
+    if (this.logLevel > LogLevel.important) {
+      return;
+    }
+    this.logger.imp(content, this.includeStackTrace);
+  }
+
+  public warn(content: string | object): void {
+    this.logger.warn(content, this.includeStackTrace);
+  }
+
+  public err(content: string | object): void {
+    this.logger.err(content, this.includeStackTrace);
+  }
+}
+
+interface ServiceLog {
+  type: string;
+  [key: string]: any;
+}
+
+/**
+ * Logs content from app services to files
+ */
+class ServiceLogger {
+  public service: string;
+  public filePath: string;
+  private stream: RotatingFileStream;
+
+  constructor(service: string, filePath: string) {
+    this.service = service;
+    this.filePath = filePath;
+    this.stream = new RotatingFileStream(this.filePath, true);
+  }
+
+  /**
+   * Converts the service log to logfmt and writes it to `this.filePath`
+   * @param serviceLog Service log object
+   * @param logToConsole True to log to console in addition to file
+   */
+  public log(serviceLog: ServiceLog, logToConsole?: boolean): void {
+    const now = new Date();
+    // RFC1123 timestamp
+    const message = `[${now.toUTCString()}] service=${this.service} ${ServiceLogger.format(serviceLog)}`;
+    this.stream.write(util.format(message, '\n'));
+    if ((LOG_LEVEL == LogLevel.debug) || logToConsole) {
```

Comment:
I don't see why not in this case

Reply:
Because you specifically asked to send logs to a file?  Why would changing the log level change where logs get sent when you didn't change the settings about where to send logs?  Just seems unintuitive to me.  But I don't feel strongly about this one

Reply:
Oh yeah i see what you are saying though it is an additional output so it goes both to console and files. I find it helpful to surface them in the console but agree it could be a bit confusing. Though seems  like it shouldn't cause any harm

Reply:
Oh, I actually misunderstood what was happening here last time.  I thought logToConsole was derived from the config file, and was the inverse of logToFiles.  But actually it's passed in at each call site (defaults to false and never used as true anywhere currently) as a way to include messages from the events and metrics log streams into the console.  Interesting.  That actually does make some sense then why requesting higher log verbosity could cause metrics and event logs to show up in the console.  I still feel a bit like it's a conflation of concepts though - changing the amount of things being logged shouldn't cause a change to where they get logged, or cause logs to be duplicated into multiple output streams, but I do see how it could be useful for quick interactive debugging sessions.  I wonder if we should instead have a separate config options `sendAllLogsToConsole` or similar that causes all the different log file streams to also get sent to the console instead of just the diagnostic log.  It's a useful mode to be able to run in, I'm just not sure it makes sense to bundle that behavior with the log verbosity control

Reply:
To be clear, feel free to punt on this for now!  This isn't a blocker by any means